examples/platforms/nrf52840/alarm.c

```diff
@@ -563,3 +572,34 @@ void RTC_IRQ_HANDLER(void)
         }
     }
 }
+
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+void otPlatTimeSet(uint64_t aTime, uint8_t aTimeSyncSeq)
+{
+    sNetworkTimeOffset = (int64_t)aTime - (int64_t)nrf5AlarmGetCurrentTime();
+    sTimeSyncSeq       = aTimeSyncSeq;
+}
+
+uint64_t otPlatTimeGet(void)
+{
+    return nrf5AlarmGetCurrentTime() + sNetworkTimeOffset;
+}
+
+void otPlatTimeIncrementSyncSeq(void)
```

Comment:
I'd like to explore ways to minimize changes to the platform APIs.  In this case, can we just pass the necessary information with the radio frame struct when sending/receiving the message?